### PR TITLE
Semicircular radial chart

### PR DESCRIPTION
**Pull request by shaswat-indian**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently the radial chart only renders a complete circle.

Issue Number: N/A

## What is the new behavior?
It now supports rendering a semicircular radial chart by setting the newly added `isSemiCircle` property to `true`, which is `false` by default.

https://github.com/reaviz/reaviz/assets/33908100/0225d932-52a1-4adc-810d-5befdf2b7a50

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/RadialAreaChart/RadialAreaChart.story.tsx [hunk 1]
```diff
@@ -100,3 +100,65 @@ export const MultiSeries = () => (
...
+                  <GradientStop offset="80%" stopOpacity={0.3} key="stop" />
+                ]}
+              />
+            }
+          />
+        }
+      />
+    }
+    axis={<RadialAxis type="category" />}
+    isSemiCircle={true}
```
Comment (amcdnl) [inline src/RadialAreaChart/RadialAreaChart.story.tsx:162]: What if we did 

```
  /**
   * Start angle for the first value.
   */
  startAngle?: number;

  /**
   * End angle for the last value.
   */
  endAngle?: number;
```

which makes it more flexible and matches other APIs in the app.

File: src/common/utils/position.ts [hunk 2]
```diff
@@ -28,7 +29,7 @@ const scaleBandInvert = (scale) => {
 /**
  * Given a point position, get the closes data point in the dataset.
  */
-export const getClosestPoint = (pos: number, scale, data, attr = 'x') => {
+export const getClosestPoint = (pos: number, scale, data, attr = 'x', isPointScale=false) => {
```
Comment (amcdnl) [inline src/common/utils/position.ts:32]: Can you add some context to what this is for ? Maybe just above the line where its used.

**Review by amcdnl — CHANGES_REQUESTED** [review]
looks great overall - minor api nit

File: src/RadialAreaChart/RadialAreaChart.story.tsx (hunk 1 above)
Comment (shaswat-indian) [reply src/RadialAreaChart/RadialAreaChart.story.tsx:162]: That would require changes in other parts wherever we have used the scales and transformations but the major challenge would  be in drawing a path for the [axis](https://github.com/reaviz/reaviz/pull/142/files#diff-1023bf738c818131abcc1515e0edb35829044401dabe4c792d6ae97a944c423aR59-R63)

File: src/common/utils/position.ts (hunk 2 above)
Comment (shaswat-indian) [reply src/common/utils/position.ts:34]: added in docstring

File: src/RadialAreaChart/RadialAreaChart.story.tsx (hunk 1 above)
Comment (amcdnl) [reply src/RadialAreaChart/RadialAreaChart.story.tsx:162]: I defiantly want to make it where we can set the start/end angle. That will make it way more flexible and allow us to use other angles in other projects.

Comment (shaswat-indian) [reply src/RadialAreaChart/RadialAreaChart.story.tsx:162]: Closing this PR. Opened https://github.com/reaviz/reaviz/pull/144 with the changes for start/end angles.

**Comment by shaswat-indian** [comment]
Closing this PR. Opened https://github.com/reaviz/reaviz/pull/144 with the changes for generic start/end angles.

**Pull request closed without merging**